packages/core/src/NodeExecuteFunctions.ts

```diff
@@ -989,7 +989,7 @@ export async function requestOAuth2(
 
 	// @ts-ignore
 	if (oAuth2Options?.keyToIncludeInAccessTokenHeader) {
-		Object.assign(newRequestOptions.headers, {
+		Object.assign(newRequestOptions.headers ?? {}, {
```

Comment:
This shuts up the compiler, but the code actually implies that if `newRequestOptions.headers` is nullish, then extend an empty object with the following values. We should instead make sure that `newRequestOptions.headers` isn't nullish before these lines are reached.

Reply:
something like this perhaps 
![image](https://user-images.githubusercontent.com/196144/192799870-9a56601b-3280-4898-8e94-514ac1981911.png)